src/Nethermind/Nethermind.Db/FullPruning/FullPruningDb.cs

```diff
@@ -198,12 +201,34 @@ public PruningContext(FullPruningDb db, IDb cloningDb, bool duplicateReads)
             public byte[]? this[ReadOnlySpan<byte> key]
             {
                 get => CloningDb[key];
-                set => _db.Duplicate(CloningDb, key, value);
+                set
+                {
+                    if (!_batches.TryDequeue(out IBatch currentBatch))
+                    {
+                        currentBatch = CloningDb.StartBatch();
+                    }
+
+                    _db.Duplicate(currentBatch, key, value);
+                    long val = Interlocked.Increment(ref _counter);
+                    if (val % 10000 == 0)
```

Comment:
Did you experiment with different values?

Reply:
Yes, not much different though. The copy rate is about 500k/s.